src/_webp.c

```diff
@@ -450,12 +450,16 @@ _anim_decoder_get_next(PyObject *self) {
     int timestamp;
     PyObject *bytes;
     PyObject *ret;
+    ImagingSectionCookie cookie;
     WebPAnimDecoderObject *decp = (WebPAnimDecoderObject *)self;
 
+    ImagingSectionEnter(&cookie);
     if (!WebPAnimDecoderGetNext(decp->dec, &buf, &timestamp)) {
+        ImagingSectionLeave(&cookie);
         PyErr_SetString(PyExc_OSError, "failed to read next frame");
         return NULL;
     }
+    ImagingSectionLeave(&cookie);
```

Comment:
```suggestion
    int ok;
    PyObject *bytes;
    PyObject *ret;
    ImagingSectionCookie cookie;
    WebPAnimDecoderObject *decp = (WebPAnimDecoderObject *)self;

    ImagingSectionEnter(&cookie);
    ok = WebPAnimDecoderGetNext(decp->dec, &buf, &timestamp)
    ImagingSectionLeave(&cookie);
    if (!ok) {
        PyErr_SetString(PyExc_OSError, "failed to read next frame");
        return NULL;
    }
```
Rather than writing `ImagingSectionLeave` twice, this follows a pattern from earlier in the file -
https://github.com/python-pillow/Pillow/blob/6782a07b8e404271dbd2e5ddf5fbb93e575ed2bc/src/_webp.c#L665-L670